### PR TITLE
LP-2858 Integrate Webinar Delete & Publish functionality on Webinar Admin Page

### PR DESCRIPTION
**Pull request by javeriaejaz-arbisoft**
[LP-2858](https://philanthropyu.atlassian.net/browse/LP-2858)

[Theme PR](https://github.com/OmnipreneurshipAcademy/adg-edx-theme/pull/523)

***

#### Description:

- Added APIView for Webinar Publish & Delete

### CODE REVIEW

**Review by annas-usman — CHANGES_REQUESTED** [review]

File: openedx/adg/admin_dashboard/webinars/api/views.py [hunk 1]
```diff
@@ -59,3 +87,27 @@ def delete(self, request, pk):
         webinar_to_delete = get_object_or_404(Webinar, pk=pk)
         webinar_to_delete.delete()
         return Response({}, status=HTTP_200_OK)
+
+
+class WebinarPublishView(APIView):
+    """
+    View to clone a webinar
+    """
+    permission_classes = (IsAuthenticated,)
```
Comment (asadali145) [inline openedx/adg/admin_dashboard/webinars/api/views.py:96]: I think we should create a new permission as well which will check whether a user is admin or not.

File: openedx/adg/lms/webinars/models.py [hunk 2]
```diff
@@ -271,13 +260,27 @@ def clean(self):
...
+    def publish(self):
+        """
+        Publish webinar on zoom if its not published
+        """
+        if not self.is_published:
+            response = ZoomClient().create_webinar(self)
+            self.meeting_link = response.get('join_url')
+            self.zoom_id = response.get('id')
+            self.is_published = True
+            self.save()
```
Comment (asadali145) [inline openedx/adg/lms/webinars/models.py:282]:
```suggestion
        if self.is_published:
            return

        response = ZoomClient().create_webinar(self)
        self.meeting_link = response.get('join_url')
        self.zoom_id = response.get('id')
        self.is_published = True
        self.save()
```

File: openedx/adg/admin_dashboard/webinars/api/views.py [hunk 3]
```diff
@@ -4,19 +4,20 @@
...
+from openedx.adg.common.lib.zoom_client.client import ZoomClient
 from openedx.adg.lms.webinars.models import Webinar
 
 
 class WebinarCloneView(APIView):
     """
     View to clone a webinar
     """
-    permission_classes = (IsAuthenticated,)
+    permission_classes = (IsADGAdmin,)
```
Comment (asadali145) [inline openedx/adg/admin_dashboard/webinars/api/views.py:20]: I think we should keep `IsAuthenticated`. What if unauthenticated user tries to hit the API, Exception will be raised as per my understanding that anonymous user has no attribute `is_active`.

Comment (javeriaejaz-arbisoft) [reply openedx/adg/admin_dashboard/webinars/api/views.py:20]: I checked, its not raising.

File: openedx/adg/admin_dashboard/permissions.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+class IsADGAdmin(permissions.BasePermission):
+    """
+    Custom permission to only allow adg admin to access admin_dashboard views.
+    """
+
+    def has_permission(self, request, view):
+        """
+        Checks if a user is active and an ADG admin or BU Admin or Superuser.
+        Failure to be any type of admin fails the condition and false is return
```
Comment (annas-usman) [inline openedx/adg/admin_dashboard/permissions.py:17]: is returned*

**Pull request merged**